### PR TITLE
Add quickfix cpp

### PR DESCRIPTION
**Pull request by dillonroach**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/quickfix-cpp/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+c_stdlib:
+  - sysroot                # [linux]
```
Comment (xhochy) [inline recipes/quickfix-cpp/conda_build_config.yaml:2]:
```suggestion
```

File: recipes/quickfix-cpp/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+c_stdlib_version:
+  - 2.17                   # [linux]
+cxx_standard:
+  - 11                     # [linux]
```
Comment (xhochy) [inline recipes/quickfix-cpp/conda_build_config.yaml:6]:
```suggestion
```

File: recipes/quickfix-cpp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+{% set version = "40b8984" %}
```
Comment (xhochy) [inline recipes/quickfix-cpp/meta.yaml:1]: Before we merge, this must be set to a real version.

File: recipes/quickfix-cpp/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    - m4
+    - autoconf
+    - automake
+    - make
+    - libxml2
+  host:
+    - libxml2
+    - openssl
+    - ruby
+    - mysql-server
```
Comment (xhochy) [inline recipes/quickfix-cpp/meta.yaml:31]: Do you really need the server in or only the library?

File: recipes/quickfix-cpp/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+requirements:
+  build:
+    - {{ stdlib('c') }}
+    - {{ compiler('cxx') }}
+    - libtool
+    - m4
+    - autoconf
+    - automake
+    - make
+    - libxml2
```
Comment (xhochy) [inline recipes/quickfix-cpp/meta.yaml:26]: Probably only needs to be in host 
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/quickfix-cpp/meta.yaml (hunk 3 above)
Comment (dillonroach) [reply recipes/quickfix-cpp/meta.yaml:1]: Yea.. fingers crossed, last official pin was 2018, but the repo is actively still releasing updates/commits - added a request here: https://github.com/quickfix/quickfix/issues/596

File: recipes/quickfix-cpp/meta.yaml (hunk 4 above)
Comment (dillonroach) [reply recipes/quickfix-cpp/meta.yaml:30]: That's a good question - was loosely basing my assumptions on https://github.com/quickfix/quickfix/blob/master/scripts/ubuntu_12/install_deps.sh - will look to eval.

File: recipes/quickfix-cpp/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
+{% set version = "40b8984" %}
+
+package:
+  name: quickfix-cpp
```
Comment (isuruf) [inline recipes/quickfix-cpp/meta.yaml:4]: Use `libquickfix` instead of `quickfix-cpp`?

File: recipes/quickfix-cpp/meta.yaml (hunk 4 above)
Comment (dillonroach) [reply recipes/quickfix-cpp/meta.yaml:31]: based on author's suggested pattern, I'd say the server needs to be there

**Comment by dillonroach** [comment]
Hi - I've updated to the latest (3mo old) commit on the repo and recommend we go with a .postYYYMMDD pattern on the version, since the package author hasn't responded to requests to get them officially designated. Happy to adjust as others see fit.

File: recipes/quickfix-cpp/meta.yaml (hunk 3 above)
Comment (dillonroach) [reply recipes/quickfix-cpp/meta.yaml]: Updated to '1.15.1.postYYYYMMDD' format, given lack of movement on official tags from package author

**Review by isuruf — APPROVED** [review]

**Comment by kcpevey** [comment]
@xhochy it looks like this is ready to cross the finish line. Can you have another look when you get a chance?

**Pull request merged**